### DISCUSSION TITLE
Quarto Dashboard Requests: Navbar Global Inputs with Modularized Pages

### DISCUSSION THREAD

**Discussion by mrworthington** [quarto-dev/quarto-cli#9942] [Q&A]
### Description

One goal of a dashboard could be to have multiple pages across a single Quarto Dashboard be controlled by a single, global input. 

I know this experience can be achieved in some formats on the front-end with a sidebar input, but my understanding is that the current version of Quarto Dashboards has two limitations:

1. **Navbar Global Inputs Issue**: Inputs can only be placed in a horizontal toolbar, a vertical sidebar, or as inputs in a dashboard card. At times, the navbar feels like wasted space and could benefit from having global dashboard inputs be placed in the navbar of dashboards with multiple pages.
2. **Modularized Pages Issue**: Dashboards with global inputs that interact across multiple pages, my understanding of the current implementation of [Quarto Dashboard Pages](https://quarto.org/docs/dashboards/layout.html#pages) is that:
     - Global inputs can be applied through a sidebar input system.
     - Pages can only be implemented through Level 1 Markdown Heading Syntax.
     
I know the current release is the first iteration of Dashboards, but I'm wondering if we can eventually manage the relationship between pages + global inputs through individual qmd docs + a .yml file. And then offering the sidebar + navbar input layouts as options for controlling global inputs.

**Comment by mcanouil** [comment]
What do you mean by "inputs" exactly? Documents? Or JavaScript/HTML component such as select menu/box, etc?

**Reply by mrworthington** [reply]
The latter. The select, range, dropdown type JavaScript/HTML inputs.

**Reply by mcanouil** [reply]
You can use code cells to add this or directly using Observable, but yes currently navigation bar is static and limited to https://quarto.org/docs/dashboards/layout.html#navigation

**Reply by mrworthington** [reply]
Thanks. To clarify, I am aware of and already using ojs code cells on Quarto Dashboards to implement inputs. One part of my question was whether these could be implemented in the navigation bar to help save space in the UI.

**Reply by mcanouil** [reply]
> One part of my question was whether these could be implemented in the navigation bar to help save space in the UI.

Not currently.
The navigation bar for dashboard project is derived from the website project which does not allow this (or in a very cumbersome way if even possible).

**Reply by mrworthington** [reply]
Got it. And what about the other one? Is it possible to implement a global input that works across multiple quarto dashboard pages with each page being its own qmd? Like, is there a way or roadmap to having a yml file that allows the global inputs to talk across multiple qmd files--with each file defining different pages of the dashboard?

**Reply by mcanouil** [reply]
> Is it possible to implement a global input that works across multiple quarto dashboard pages with each page being its own qmd?

yes (in the same model as Website or Book projects), it is possible but unlikely to happen soon.
But you can achieve something close using the include shortcode (no YAML though).

**Reply by jimjam-slam** [reply]
Dashboards are a single docs, so maybe a Lua filter could relocate marked OJS cells to the nav bar? I feel like it might require some additional styling to ensure they don't break out of the linked space up there, but...

**Reply by mcanouil** [reply]
Indeed, in the end it's a single HTML file so you can do advanced post-processing in the file or in the browser via JavaScript, but it might lead to unintended side effects.
